gc.c

```diff
@@ -1582,6 +1582,17 @@ os_obj_of(VALUE of)
  *     2.2250738585072e-308
  *     Total count: 7
  *
+ *  When the process is in multi-Ractor mode (when <code>Ractor.new</code>
+ *  has been called within the process at least once), this method yields
+ *  Ractor-shareable objects only.
```

Comment:
I think we should clarify this is a bug (it's clearly incompatible), it just hasn't been solved yet.
And maybe link to one of these issues.

Reply:
Also https://bugs.ruby-lang.org/issues/19387 is opened so it's not just "won't fix" and a known bug.

Reply:
Understood. I made it clear that the current behavior is not ideal in 00bdbbaea8bdfa690a12cf78aad18d9c3f8cdb62. Does this look good to you?